src/k8s/pkg/k8sd/features/cilium/gateway.go

```diff
@@ -81,6 +74,19 @@ func ApplyGateway(ctx context.Context, snap snap.Snap, gateway types.Gateway, ne
 		}
 	}
 
+	// Remove Gateway CRDs if the Gateway feature is disabled.
+	// This is done after the Cilium update as cilium requires the CRDs to be present for cleanups.
+	if !gateway.GetEnabled() {
+		if _, err := m.Apply(ctx, chartGateway, helm.StateDeleted, nil); err != nil {
+			err = fmt.Errorf("failed to delete Gateway API CRDs: %w", err)
+			return types.FeatureStatus{
+				Enabled: false,
+				Version: ciliumAgentImageTag,
+				Message: fmt.Sprintf(gatewayDeleteFailedMsgTmpl, err),
+			}, err
+		}
+	}
+
 	if !changed {
```

Comment:
if the enable/disable code paths start drifting, better move them into separate enableGateway and disableGateway functions